### ISSUE TITLE
Mosaic terminology

### ISSUE THREAD

**Issue by datakurre** [plone/plone.app.mosaic#125]
@agitator @pbauer Since you are probably presenting Mosaic at the Conference...

Are we all clear on Mosaic terminology and is there anything we should fix in it?

* site layout (default site layout, section site layout, page site layout)
* panel
* content layout (aka page layout)
* tile (text tiles, field tiles, app tiles)

Site layout replaces main template. It was supposed to be so simple that a site section or even a single page could have its own site layout (section site layout or page site layout), if required (and that does not result similar chaos to customizing main_template, because site layouts are centrally managed in plone.resource directories). Yet, because users still expect Plone to look Plone, the default site layouts look pretty messy, and we are mostly relying on bbb magic, which creates default site layout from main_template.

Panel is a named customizable area in site layout, not completely different from metal:slot. Site layout can define any amount of panels. Main template generated site layout has only one panel, named "content". To make things easier, we assume "content" panel to always exist and there's even some magic code to map "content" panel into "#content" element when it does not exist.

Content layout replaces browser view templates. It contains reference for required site layout and content instance specific layout in one or more named panel elements. (In the original Deco vision there was going to be only one content type called "Page" and therefore "content layout" is still referred as "page layout" in some places; more generic "content layout" term was invented in Barcelona Mosaic Sprint.) When content layout is rendered, site layout is fetched and matching content layout's panels are merged into similarly named panels in site layout. All layout outside panels in content layout is dropped.

Tile is a special browser view template referred in layouts and rendered and merged into them after panels are merged. Some systems call tiles "blocks", but that might be confusing in Plone, because the layout engine for merging layouts and tiles is called "blocks".

Content in content layout panels is usually contained in tiles to make them editable in Mosaic layout editor. The editor has very verbose grid syntax. Text tiles are not "plone.tiles", but just HTML within the layout HTML. Field tiles are all "plone.app.standardtiles.field" tiles with different configuration. All the rest "plone.tiles" tiles are called application tiles.

Yet, because we seem to be hesitating to move to use site layouts (because main_template based bbb layer works too well), we are introducing one more layer into Mosaic: editors can define such "content type view" layouts outside content instances that they are merged into site layout before  the content instance layout is merged. And content instance layout is merged only after that. In practice, this would allow such named views in "Display menu", that changing view would alter how content instance specific layouts are rendered, e.g. changing from 1 panel layout to 2 panels layout or multi row layout or anything while preserving editor made panel layouts.

Is this all ok? Any ideas to make these more clear. How to explain the last one better?
